### ISSUE TITLE
Feature Request: Change error message text and the case to show it - reported by @Puneet-here

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#10772]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
## Problem: 
We are showing Please enter a valid phone number without brackets or dashes. If you\'re outside the US please include your country code, eg. [+447782339811](tel:+447782339811) just after entering one digit. This feels weird in a situation where a user enters a digit to search for a recent contact or searching an email or name which contains that number.

## Solution: 
On adding two numbers we add the country code to the phone number. So shouldn't we just say Please enter a valid phone number without brackets or dashes.  and only show this when a user types a phone number in a wrong format (uses brackets/dashes)

## Action Performed:
1. Open app and navigate to search 
2. type any digit 
3. Check message under search CTA 

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.96-0
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 
<img width="372" alt="Screenshot 2022-07-02 at 11 46 54 PM" src="https://user-images.githubusercontent.com/43995119/188055570-16072a95-415b-46ce-b5fb-4973a3e725d5.png">

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @Puneet-here
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1660969852572289

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by varshamb** [comment]
### Proposal

We need to tweak validation in below to check entered value is single number.

https://github.com/Expensify/App/blob/27b7f652eafb772b559488e61c3e016e760cff75/src/libs/OptionsListUtils.js#L800

**Comment by Puneet-here** [comment]
### Proposal 

We have 2 options -

1. Remove the error message below the search completely because when user tries to create a chat with invalid number we shows an error. And why not just allow user to search a number with hyphens, brackets in it (we allow it at other places). 

2. Change the validation here. This way we won't see this message when we aren't actually searching for a phone number, we will only see it when we use brackets and hyphen in a number, at that time we can suggest to not use hyphen and brackets by showing this error.

https://github.com/Expensify/App/blob/27b7f652eafb772b559488e61c3e016e760cff75/src/libs/OptionsListUtils.js#L805

We can use a regex to check if the search term contains hyphen/bracket. If it does then we will show the error. For example this one - `/[-()]/`

```js 
    if (searchValue && CONST.REGEX.HYPHEN_AND_BRACKET.test(searchValue))
```

https://user-images.githubusercontent.com/101883770/188173349-39c23a9c-9eb9-4ead-9f8c-b05014588490.mov




At least we should remove this part from the error `If you're outside the US please include your country code, eg. +447782339811` because when a user (who resides outside of US) searches a number we are adding the country code ourself so the user can't actually search a number without the country code
<img width="367" alt="Screenshot 2022-09-02 at 1 22 01 PM" src="https://user-images.githubusercontent.com/101883770/188086704-5c25f0fe-a61b-4feb-9cc6-66e6b80f30af.png">

**Comment by cdraeger11** [comment]
Looks good to me!

**Comment by miljakljajic** [comment]
already triaged - adding engineering

**Comment by neil-marcellini** [comment]
I agree that this should be fixed so that it only shows when you have entered an invalid phone number with brackets or dashes. Also, this is error text so I think the form should be outlined in red and the text should also be red. We should follow the error guidelines is [FORMS.md](https://github.com/Expensify/App/blob/8e07c62fb9780c627106f45eb72614466a850f27/contributingGuides/FORMS.md#L124-L128). Sending this off as an external issue.

**Comment by Puneet-here** [comment]
I am not sure about showing it as red and making the field red because this is a hint which we just need to show when user searches an invalid phone, so the user will know that they shouldn't use hyphen and bracket and if the user still searches a number like this and tries to create a chat then we are already showing an error.

**Comment by laurenreidexpensify** [comment]
https://www.upwork.com/jobs/~01dd6a1b05300dd153

**Comment by Puneet-here** [comment]
Hi @marcaaron, this issue contains two things:-

1. The case when we show hint message:- After entering only a digit in search we see the hint message saying: `Please enter a valid phone number without brackets or dashes. If you are outside the US please include your country code (e.g. +15005550006).` (refer to the Screenshot below)
The problem is that this message shouldn't be shown only just after adding one digit in the search because the user can be searching for a name/email or an already added contact. So instead we should show this hint message when they add brackets or dashes in the number. It can be done easily by using regex to check number. 

<img width="373" alt="Screenshot 2022-09-15 at 7 13 15 PM" src="https://user-images.githubusercontent.com/101883770/190419619-a0e72da7-9c38-45ec-a36a-ac7fa71a67ce.png">

2. The hint message text:- The hint message says `If you are outside the US please include your country code (e.g. +15005550006).` this line is not needed because if you are outside of US on searching a number the country code will be automatically added so there's no way for you to search a number without the country code. So we should remove this line.

**Comment by Puneet-here** [comment]
@marcaaron, can you check the above comment I have explained the issue.

**Comment by marcaaron** [comment]
>instead we should show this hint message when they add brackets or dashes in the number.

Not sure if this logic makes sense. There's no way to tell whether someone is entering a non-phone number or phone number without dashes or brackets. 

So it seems reasonable to show the hint as long as there are no characters that would indicate this is an email (alphabet or `@` symbols should remove the hint).

>if you are outside of US on searching a number the country code will be automatically added 

Sorry, didn't understand this. Where does the country code get added automatically?

>there's no way for you to search a number without the country code. So we should remove this line.

What do you mean by this?

**Comment by Puneet-here** [comment]
I haven't look into the code so I can't link the code where the country code gets added but here's the video

https://user-images.githubusercontent.com/101883770/199331784-c168e4e7-bc94-4553-9c78-530da93bebb6.mov

**Comment by Puneet-here** [comment]
> What do you mean by this?

The message says `If you are outside the US please include your country code (e.g. +15005550006)` but as shown in the video you can't search a number without country code if you are outside of US

**Comment by marcaaron** [comment]
>I haven't look into the code so I can't link the code where the country code gets added but here's the video

Seems weird. I'm not sure what the expected behavior should be here, but feels like adding the country code automatically is a different bug. How are you supposed to start a conversation with someone who is not in your country?

>you can't search a number without country code if you are outside of US

This also seems like a bug. If you have a chat with someone I would expect that search would show you any chats that match on any part of a phone number.

**Comment by Puneet-here** [comment]
> How are you supposed to start a conversation with someone who is not in your country?

You can add the country code yourself, so that isn't a problem

> This also seems like a bug. If you have a chat with someone I would expect that search would show you any chats that match on any part of a phone number.

Yeah it does that but it also starts showing a contact with country code

<img width="517" alt="Screenshot 2022-11-02 at 2 46 37 AM" src="https://user-images.githubusercontent.com/101883770/199343337-581005e2-4e68-4305-9e26-4b7cec3c3b5f.png">

I think we should also remove this line `If you are outside the US please include your country code (e.g. +15005550006)` since it doesn't make any sense. A user outside of US can't search a number without country code.

**Comment by marcaaron** [comment]
Fixing the copy doesn't help much if the whole feature is off. Anyway, this needs more feedback from others and for someone to lead the improvements. IMO we shouldn't prevent anyone from searching for a phone number that exists or creating a chat with a specific number (regardless of country code or current location). I think where we are going wrong here is superimposing the country code at all. If we are going to fix this copy it should happen when we improve the feature itself.
